drs_downloader_to_getm.wdl

```diff
@@ -172,11 +179,17 @@ task download {
             start_time=`date +%s`
             signed_urls=($(cat ~{manifest} | jq -r '.[] .url'))
             for signed_url in ${signed_urls[@]}; do
-                # this is going to create some crazy truncated names but it shouldn't make a difference in run times
-                curl ${signed_url} -P ${TMP_DL_DIR}/
+                curl ${signed_url} -o ${TMP_DL_DIR}/$(basename $signed_url)
```

Comment:
Running:
```
basename "https://storage.googleapis.com/topmed-irc-share/genomes/NWD314407.b38.irc.v1.cram.crai?GoogleAccessId=mbaumann1-989@dcpstage-210518.iam.gserviceaccount.com&Expires=1613020693&Signature=SEaop7DYgR%2BVDT8C%2B72PcSyBA2FJmIJ8%2BnY9AvmE%2BtlTTOQ0vITqr%2FVWFt60sPPEahEgK324af7YgZk6S55l3i7bhc2uakGI19tQxV2l2HSnjqKdTU72%2Fcf8%2FG6nvuXgOQb9dmhNjIs2Gt0gLqbEzBXsYlCXCViNAUIHlVOnF0Xu8CoD8bvXHk3i97kFeqcgWD%2BKZRGu8MHqmy3BfzKcWjxD9uftDswoRFOP04w7yDvLIe%2Bm5ce%2F7aNxKpWfuYlCX%2BiYsWS0OtcsnFb0Lxt%2BhbN2v0T%2B6%2F%2BzejXAVRyBLBgR5CcqI42pk8VkW3r9%2FX0tSHrLo%2BYKgnwHkZaGZ0PBcw==&userProject=dcpstage-billing"
```
results in:
```
NWD314407.b38.irc.v1.cram.crai?GoogleAccessId=mbaumann1-989@dcpstage-210518.iam.gserviceaccount.com&Expires=1613020693&Signature=SEaop7DYgR%2BVDT8C%2B72PcSyBA2FJmIJ8%2BnY9AvmE%2BtlTTOQ0vITqr%2FVWFt60sPPEahEgK324af7YgZk6S55l3i7bhc2uakGI19tQxV2l2HSnjqKdTU72%2Fcf8%2FG6nvuXgOQb9dmhNjIs2Gt0gLqbEzBXsYlCXCViNAUIHlVOnF0Xu8CoD8bvXHk3i97kFeqcgWD%2BKZRGu8MHqmy3BfzKcWjxD9uftDswoRFOP04w7yDvLIe%2Bm5ce%2F7aNxKpWfuYlCX%2BiYsWS0OtcsnFb0Lxt%2BhbN2v0T%2B6%2F%2BzejXAVRyBLBgR5CcqI42pk8VkW3r9%2FX0tSHrLo%2BYKgnwHkZaGZ0PBcw==&userProject=dcpstage-billing
```
which is a very wonky filename. I suppose as long as the shell treats it as a string, and does not try to evaluate it and is therefore is not confused by the `&` characters, etc.
This may work.
I don't know if this is what you intended or not.
Again, you could instead use the `filename` value from the getm manifest as the name of the output file, if the above is what you intended but it doesn't work.

Reply:
Actually, it doesn't work as it stands, because the resulting filename is too long (for Linux and MacOS anyway):

```
$ touch $(basename "https://storage.googleapis.com/topmed-irc-share/genomes/NWD314407.b38.irc.v1.cram.crai?GoogleAccessId=mbaumann1-989@dcpstage-210518.iam.gserviceaccount.com&Expires=1613020693&Signature=SEaop7DYgR%2BVDT8C%2B72PcSyBA2FJmIJ8%2BnY9AvmE%2BtlTTOQ0vITqr%2FVWFt60sPPEahEgK324af7YgZk6S55l3i7bhc2uakGI19tQxV2l2HSnjqKdTU72%2Fcf8%2FG6nvuXgOQb9dmhNjIs2Gt0gLqbEzBXsYlCXCViNAUIHlVOnF0Xu8CoD8bvXHk3i97kFeqcgWD%2BKZRGu8MHqmy3BfzKcWjxD9uftDswoRFOP04w7yDvLIe%2Bm5ce%2F7aNxKpWfuYlCX%2BiYsWS0OtcsnFb0Lxt%2BhbN2v0T%2B6%2F%2BzejXAVRyBLBgR5CcqI42pk8VkW3r9%2FX0tSHrLo%2BYKgnwHkZaGZ0PBcw==&userProject=dcpstage-billing")
touch: cannot touch 'NWD314407.b38.irc.v1.cram.crai?GoogleAccessId=mbaumann1-989@dcpstage-210518.iam.gserviceaccount.com&Expires=1613020693&Signature=SEaop7DYgR%2BVDT8C%2B72PcSyBA2FJmIJ8%2BnY9AvmE%2BtlTTOQ0vITqr%2FVWFt60sPPEahEgK324af7YgZk6S55l3i7bhc2uakGI19tQxV2l2HSnjqKdTU72%2Fcf8%2FG6nvuXgOQb9dmhNjIs2Gt0gLqbEzBXsYlCXCViNAUIHlVOnF0Xu8CoD8bvXHk3i97kFeqcgWD%2BKZRGu8MHqmy3BfzKcWjxD9uftDswoRFOP04w7yDvLIe%2Bm5ce%2F7aNxKpWfuYlCX%2BiYsWS0OtcsnFb0Lxt%2BhbN2v0T%2B6%2F%2BzejXAVRyBLBgR5CcqI42pk8VkW3r9%2FX0tSHrLo%2BYKgnwHkZaGZ0PBcw==&userProject=dcpstage-billing': File name too long
```